solr/core/src/java/org/apache/solr/util/circuitbreaker/RequestType.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.util.circuitbreaker;
+
+/** Request types that can be registered for a {@link CircuitBreaker} */
```

Comment:
Adding this classification for request types is great and probably could have plenty of usages, not only for circuit breaker.
I suggest this is moved to a more generic package.

Reply:
Good point. We already have `SolrRequest.SolrRequestType` which fits nicely. Will switch to using it.